pkg/kubectl/cmd/config/use_context.go

```diff
@@ -82,10 +84,16 @@ func (o *useContextOptions) complete(cmd *cobra.Command) bool {
 	return true
 }
 
-func (o useContextOptions) validate() error {
+func (o useContextOptions) validate(config *clientcmdapi.Config) error {
 	if len(o.contextName) == 0 {
 		return errors.New("You must specify a current-context")
 	}
 
-	return nil
+	for _, context := range config.Contexts {
+		if context.Cluster == o.contextName {
```

Comment:
> Ok that's confusing. I thought we were using NamedContext but it looks like that's in the v1 clientcmd api

We weren't supposed to serialize maps in external packages, so externally it's a slice of NamedContext.  Internally, we're looking for exactly the behavior of a map (lookup by key, no duplicate keys allowed) so that's the internal representation.  Internal came first and I got my hand smacked for serializing the map.